internal/network_client.go

```diff
@@ -24,10 +24,27 @@ type NetworkClient interface {
 
 	// GetAccountDetails retrieves account information for the given account ID.
 	GetAccountDetails(accountID string) (*AccountDetails, error)
+
+	// SimulateTransaction simulates a transaction and returns the result.
+	// This is required for Soroban transactions to get resource fees and auth entries.
+	// For network clients that don't support simulation (like Horizon), this returns an error.
+	SimulateTransaction(txXDR string) (*SimulateTransactionResult, error)
+
+	// SupportsContractAddresses returns true if this network client can fund
+	// contract addresses (C addresses). RPC supports this, Horizon does not.
+	SupportsContractAddresses() bool
```

Comment:
posted comment above for consideration on [removal of SupportsContractAddresses](https://github.com/stellar/friendbot/pull/21/changes#r2612833658)